app/factory/Layer.js

```diff
@@ -226,21 +247,69 @@ Ext.define('CpsiMapview.factory.Layer', {
         Ext.log.info('Not implemented yet', layerConf, layerType);
     },
 
+    /**
+     * Creates a World Wind (BlueMarble-200412) layer
+     *
+     * @param  {Object} layerConf  The configuration object for this layer
+     * @return {ol.layer.Tile}     World Wind (BlueMarble-200412) layer
+     */
     createNasa: function(layerConf) {
+        var nasaWms = this.createWms({
+            url: 'https://worldwind25.arc.nasa.gov/wms?',
+            serverOptions: {
+                layers: 'BlueMarble-200412'
+            },
+            openLayers: layerConf.openLayers
+        });
 
-        Ext.log.info('Not implemented yet', layerConf);
+        return nasaWms;
     },
 
     createOs: function(layerConf) {
         Ext.log.info('Not implemented yet', layerConf);
     },
 
     createArcGisCache: function(layerConf) {
+        // Maybe this helps: https://stackoverflow.com/a/41608464
         Ext.log.info('Not implemented yet', layerConf);
     },
 
+    /**
+     * Creates an ArcGIS REST layer
+     *
+     * @param  {Object} layerConf  The configuration object for this layer
+     * @return {ol.layer.Tile}     ArcGIS REST layer
+     */
     createArcGisRest: function(layerConf) {
-        Ext.log.info('Not implemented yet', layerConf);
+        var layer;
+        var singleTile = layerConf.openLayers.singleTile;
+        if (singleTile) {
+            layer = new ol.layer.Image({
+                source: new ol.source.ImageArcGISRest({
+                    url: layerConf.url,
+                    params: layerConf.serverOptions || {},
+                    ratio: 1
+                }),
+                visible: layerConf.openLayers.visibility,
+                minResolution: layerConf.openLayers.minResolution,
+                maxResolution: layerConf.openLayers.maxResolution,
+                opacity: layerConf.openLayers.opacity
+            });
+        } else {
+            layer = new ol.layer.Tile({
+                name: layerConf.text,
+                source: new ol.source.TileArcGISRest({
+                    url: layerConf.url,
+                    params: layerConf.serverOptions || {}
+                }),
+                visible: layerConf.openLayers.visibility,
+                minResolution: layerConf.openLayers.minResolution,
+                maxResolution: layerConf.openLayers.maxResolution,
+                opacity: layerConf.openLayers.opacity
+            });
+        }
+
+        return layer;
```

Comment:
Same comment wrt singleTile and duplicated code. I'll let you decide.

Reply:
See my comment in #34